### PR TITLE
scUSDS v2 | Real Yield

### PR DESCRIPTION
**Pull request by dxganta**
- There is no USDS Chainlink Oracle, so no price converter has been added. We are using the DAI-WETH PriceConverter since DAI to USDS rate is always 1:1
- The aaveV3 adapter for USDS has the claimRewards activated since there are aUSDS rewards being emitted for supplying USDS currently on AAVE V3.

### CODE REVIEW

File: src/steth/scUSDSv2.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import {ISinglePairPriceConverter} from "./priceConverter/ISinglePairPriceConverter.sol";
+import {ISinglePairSwapper} from "./swapper/ISinglePairSwapper.sol";
+
+/**
+ * @title scUSDSv2
+ * @notice Sandclock USDS Vault implementation.
+ * @dev Inherits from scCrossAssetYieldVault to manage and generate USDS yield.
+ * @dev There is no USDS Chainlink Feed, but since USDS to DAI is always 1:1 so
+ * maybe we can use the DAI Price Converter here.
+ * @dev This vault also receives aUSDS rewards which must be claimed periodically using claimRewards()
```
Comment (brkomir) [inline src/steth/scUSDSv2.sol:19]: should we integrate this into rebalance script? so for example, rebalance script for scUSDSv2 checks rewards amount and if greater than some threshold, it automatically calls the claimRewards func?

Comment (dxganta) [reply src/steth/scUSDSv2.sol:19]: Yes thats a good idea, but it needs to be done on the scripts-update pr right?

File: test/scUSDSv2.t.sol [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    scWETH wethVault = scWETH(payable(M.SCWETHV2));
+    scUSDSv2 vault;
+
+    AaveV3ScUsdsAdapter aaveV3Adapter;
+    ISinglePairSwapper swapper;
+    ISinglePairPriceConverter priceConverter;
+
+    uint256 pps;
+
+    function setUp() public {
```
[the remark this answers is not in the document]
Comment (dxganta) [reply test/scUSDSv2.t.sol]: [Done](https://github.com/lindy-labs/sandclock-contracts/pull/168/commits/24c60b8e82bff73ca4c0017b06ea3fa565003c34)

File: src/steth/scUSDSv2.sol [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import {Constants as C} from "../lib/Constants.sol";
+import {ISinglePairPriceConverter} from "./priceConverter/ISinglePairPriceConverter.sol";
+import {ISinglePairSwapper} from "./swapper/ISinglePairSwapper.sol";
+
+/**
+ * @title scUSDSv2
+ * @notice Sandclock USDS Vault implementation.
+ * @dev Inherits from scCrossAssetYieldVault to manage and generate USDS yield.
+ * @dev There is no USDS Chainlink Feed, but since USDS to DAI is always 1:1 so
+ * maybe we can use the DAI Price Converter here.
```
[the remark this answers is not in the document]
Comment (dxganta) [reply src/steth/scUSDSv2.sol]: [Fixed](https://github.com/lindy-labs/sandclock-contracts/pull/168/commits/24c60b8e82bff73ca4c0017b06ea3fa565003c34)

**Review by brkomir — APPROVED** [review]

**Pull request merged**